packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -170,34 +180,20 @@ class IOSDevice extends Device {
   @override
   bool get supportsStartPaused => false;
 
-  static Future<List<IOSDevice>> getAttachedDevices() async {
-    if (!globals.platform.isMacOS) {
+  static Future<List<IOSDevice>> getAttachedDevices(Platform platform, XCDevice xcdevice) async {
+    if (!platform.isMacOS) {
       throw UnsupportedError('Control of iOS devices or simulators only supported on Mac OS.');
     }
-    if (!globals.iMobileDevice.isInstalled) {
-      return <IOSDevice>[];
-    }
 
-    final List<IOSDevice> devices = <IOSDevice>[];
-    for (String id in (await globals.iMobileDevice.getAvailableDeviceIDs()).split('\n')) {
-      id = id.trim();
-      if (id.isEmpty) {
-        continue;
-      }
+    return await xcdevice.getAvailableIOSDevices()?.toList() ?? const <IOSDevice>[];
+  }
 
-      try {
-        final String deviceName = await globals.iMobileDevice.getInfoForDevice(id, 'DeviceName');
-        final String sdkVersion = await globals.iMobileDevice.getInfoForDevice(id, 'ProductVersion');
-        devices.add(IOSDevice(id, name: deviceName, sdkVersion: sdkVersion));
-      } on IOSDeviceNotFoundError catch (error) {
-        // Unable to find device with given udid. Possibly a network device.
-        globals.printTrace('Error getting attached iOS device: $error');
-      } on IOSDeviceNotTrustedError catch (error) {
-        globals.printTrace('Error getting attached iOS device information: $error');
-        UsageEvent('device', 'ios-trust-failure').send();
-      }
+  static Future<List<String>> getDiagnostics(Platform platform, XCDevice xcdevice) async {
+    if (!platform.isMacOS) {
```

Comment:
I changed it to:
```
return const <String>['Control of iOS devices or simulators only supported on Mac OS.'];
```
since that's the relevant diagnostic info. We don't really need to crash about gathering diagnostics.